gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/NewDagManager.java

```diff
@@ -0,0 +1,352 @@
+    for (int delay = INITIAL_HOUSEKEEPING_THREAD_DELAY; delay < MAX_HOUSEKEEPING_THREAD_DELAY; delay *= 2) {
+      houseKeepingThreadPool.schedule(() -> {
+        try {
+          loadDagFromDagStateStore();
+        } catch (Exception e ) {
+          log.error("failed to sync dag state store due to ", e);
+        }}, delay, TimeUnit.MINUTES);
+    }
+  }
+
+  public DagStateStore createDagStateStore(Config config, Map<URI, TopologySpec> topologySpecMap) {
+    try {
+      Class<?> dagStateStoreClass = Class.forName(ConfigUtils.getString(config, DAG_STATESTORE_CLASS_KEY, FSDagStateStore.class.getName()));
+      return (DagStateStore) GobblinConstructorUtils.invokeLongestConstructor(dagStateStoreClass, config, topologySpecMap);
+    } catch (ReflectiveOperationException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void loadDagFromDagStateStore() throws IOException {
+    List<Dag<JobExecutionPlan>> dags = this.dagStateStore.getDags();
+    log.info("Loading " + dags.size() + " dags from dag state store");
+    for (Dag<JobExecutionPlan> dag : dags) {
+      // todo - reloaded dags should have a new type of FlowActionType
+      this.dagTaskStream.addDagAction(DagManagerUtils.createDagActionFromDag(dag, DagActionStore.FlowActionType.LAUNCH));
+    }
+  }
+
+  public void removeDagActionFromStore(DagActionStore.DagAction dagAction) throws IOException {
+    if (this.dagActionStore.isPresent()) {
+      this.dagActionStore.get().deleteDagAction(dagAction);
+    }
+  }
+
+  //@Subscribe todo uncomment it when new dag manager is ready
+  public void handleKillFlowEvent(KillFlowEvent killFlowEvent) {
+    String flowGroup = killFlowEvent.getFlowGroup();
+    String flowName = killFlowEvent.getFlowName();
+    if (killFlowEvent.getFlowExecutionId() != null) {
+      long flowExecutionId = killFlowEvent.getFlowExecutionId();
+      log.info("Received kill request for flow ({}, {}, {})", flowGroup, flowName, flowExecutionId);
+      // todo - should this be added to the dac action store instead?
+      this.dagTaskStream.addDagAction(new DagActionStore.DagAction(flowGroup, flowName, String.valueOf(flowExecutionId), DagActionStore.FlowActionType.KILL));
+
+    } else {
+      List<Long> flowExecutionIds = getJobStatusRetriever().getLatestExecutionIdsForFlow(flowName, flowGroup, 10);
+      log.info("Found {} flows to cancel.", flowExecutionIds.size());
+      for (long flowExecutionId : flowExecutionIds) {
+        handleKillFlowEvent(new KillFlowEvent(flowGroup, flowName, flowExecutionId));
+      }
+    }
+  }
+
+  // @Subscribe todo uncomment it when new dag manager is ready
+  public void handleResumeFlowEvent(ResumeFlowEvent resumeFlowEvent) {
+    String flowGroup = resumeFlowEvent.getFlowGroup();
+    String flowName = resumeFlowEvent.getFlowName();
+    String flowExecutionId = String.valueOf(resumeFlowEvent.getFlowExecutionId());
```

Comment:
do we need a null check here like above for kill or this API always requires flowExecutionID?

Reply:
Let only one flow execution to be resumed, so let flowExecutionId be required for this API.